tests/integration/test_algorithms.py

```diff
@@ -82,7 +82,38 @@ def test_gaussian_gate_teleportation(setup_eng, pure):
     cov2 = state.reduced_gaussian(3)[1]
     assert np.allclose(cov1, cov2, atol=0.05, rtol=0)
 
+@pytest.mark.backends("bosonic")
+def test_bosonic_gate_teleportation(setup_eng, pure):
+    """Test that gaussian states match after gate teleportation"""
+    eng, prog = setup_eng(4)
```

Comment:
maybe add a comment indicating that mode `q[3]` serves as a reference?

Reply:
Done.